src/FSharpVSPowerTools.Logic/GoToDefinitionFilter.fs

```diff
@@ -421,10 +432,21 @@ type GoToDefinitionFilter(textDocument: ITextDocument,
                 match symbolResult with
                 | Some (_, _, _, _, FSharpFindDeclResult.DeclFound _) 
                 | None ->
-                    // Run the operation on UI thread since continueCommandChain may access UI components.
-                    do! Async.SwitchToContext uiContext
-                    // Declaration location might exist so let Visual F# Tools handle it. 
-                    return continueCommandChain()
+                    // no FSharpSymbol found, here we look at #load directive
```

Comment:
Regarding `Async.RunSynchronously` I'm trying to remove it:

````fsharp
                    // no FSharpSymbol found, here we look at #load directive
                    let directive = maybe {
                        let! filepath, project, _ = getCurrentFilePathProjectAndDoc()
                        return vsLanguageService.GetLoadDirectiveFileNameAtCursor(filepath, view, project)
                    }
                    let maybeFileToOpen =
                        match directive with
                        | Some maybeResultAsync ->
                            let! r = maybeResultAsync // This construct may only be used within computation expressions
                            r
                        | None -> None

                    match maybeFileToOpen with
                    | Some fileToOpen ->
                        // directive found, navigate to the file at first line
                        serviceProvider.NavigateTo(fileToOpen, 0, 0, 0, 0)
                    | None ->
                        // Run the operation on UI thread since continueCommandChain may access UI components.
                        do! Async.SwitchToContext uiContext
                        // Declaration location might exist so let Visual F# Tools handle it. 
                        return continueCommandChain()
```

although the whole thing is within `async { ... }` and not in another computation expression.

Is this me doing something incorrect or a compiler bug?